mmv1/products/discoveryengine/ChatEngine.yaml

```diff
@@ -0,0 +1,175 @@
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'ChatEngine'
+description: |
+  Vertex chat and Conversation Enging Chat type
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Vertex AI Conversation': 'https://cloud.google.com/dialogflow/cx/docs/concept/generative'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1alpha/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+update_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+async: !ruby/object:Api::OpAsync
+  actions: ['create', 'delete']
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engineId'
+    description: |
+      The ID to use for chat engine.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collectionId.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'industryVertical'
+    description: |
+      Industry Vertical
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      displayName
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    min_size: 1
+    description: |
+      Data Store IDs, Data Stores cannot be removed once added.
+      Removing a Data Store from Chat Engine will result in error.
+      To resolve error, first use `terraform refresh` command to correct state,
+      then revert the code change to ensure graceful teardown.
+    required: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Enum
+    name: 'solutionType'
```

Comment:
This field feels a little duplicative; couldn't the API infer it from the presence of `chatEngineConfig` vs the other options?

Reply:
whoops, didn't see your response. +1 to implicitly setting it - I went into how to do so in a different thread.